src/test/kotlin/org/rust/ide/refactoring/RsChangeSignatureTest.kt

```diff
@@ -0,0 +1,885 @@
+        fn foo/*caret*/(s: S<bool>) { unimplemented!() }
+                      //^
+    """, """
+        struct S<T, R=u32>(T, R);
+        fn foo/*caret*/(s: S<bool>) -> S<bool> { unimplemented!() }
+                      //^
+    """) {
+        val parameter = findElementInEditor<RsValueParameter>()
+        returnTypeDisplay = parameter.typeReference!!
+    }
+
+    fun `test remove return type`() = doTest("""
+        fn foo/*caret*/() -> u32 { 0 }
+    """, """
+        fn foo() { 0 }
+    """) {
+        returnTypeDisplay = createType("()")
+    }
+
+    fun `test remove return type without block`() = doTest("""
+        trait Trait {
+            fn foo/*caret*/() -> i32;
+        }
+    """, """
+        trait Trait {
+            fn foo() -> u32;
+        }
+    """) {
+        returnTypeDisplay = createType("u32")
+    }
+
+    fun `test remove only parameter`() = doTest("""
+        fn foo/*caret*/(a: u32) {
+            let c = a;
+        }
+        fn bar() {
+            foo(0);
+        }
+    """, """
+        fn foo() {
+            let c = a;
+        }
+        fn bar() {
+            foo();
+        }
+    """) {
+        parameters.removeAt(0)
+    }
+
+    fun `test remove first parameter`() = doTest("""
+        fn foo/*caret*/(a: u32, b: u32) {
+            let c = a;
+        }
+        fn bar() {
+            foo(0, 1);
+        }
+    """, """
+        fn foo(b: u32) {
+            let c = a;
+        }
+        fn bar() {
+            foo(1);
+        }
+    """) {
+        parameters.removeAt(0)
+    }
+
+    fun `test remove middle parameter`() = doTest("""
+        fn foo/*caret*/(a: u32, b: u32, c: u32) {
+            let c = a;
+        }
+        fn bar() {
+            foo(0, 1, 2);
+        }
+    """, """
+        fn foo(a: u32, c: u32) {
+            let c = a;
+        }
+        fn bar() {
+            foo(0, 2);
+        }
+    """) {
+        parameters.removeAt(1)
+    }
+
+    fun `test remove last parameter`() = doTest("""
+        fn foo/*caret*/(a: u32, b: u32) {}
+        fn bar() {
+            foo(0, 1);
+        }
+    """, """
+        fn foo(a: u32) {}
+        fn bar() {
+            foo(0);
+        }
+    """) {
+        parameters.removeAt(parameters.size - 1)
+    }
+
+    fun `test remove last parameter trailing comma`() = doTest("""
+        fn foo/*caret*/(a: u32,) {}
+    """, """
+        fn foo() {}
+    """) {
+        parameters.clear()
+    }
+
+    fun `test add only parameter`() = doTest("""
+        fn foo/*caret*/() {}
+        fn bar() {
+            foo();
+        }
+    """, """
+        fn foo(a: u32) {}
+        fn bar() {
+            foo();
+        }
+    """) {
+        parameters.add(parameter("a", createType("u32")))
+    }
+
+    fun `test add last parameter`() = doTest("""
+        fn foo/*caret*/(a: u32) {}
+        fn bar() {
+            foo(0);
+        }
+    """, """
+        fn foo(a: u32, b: u32) {}
+        fn bar() {
+            foo(0, );
```

Comment:
Maybe we should add `todo!()` in function calls for added parameters?

Reply:
In my opinion it is better to notify the user about the invalid call sites immediately (by making the code fail to compile), so that they can be fixed ASAP. I wouldn't want to only find about it later at runtime. But I suppose that this can be solved with the default parameter values, if we implement them later? We can default these to `todo!()` possibly.

Reply:
> it is better to notify the user about the invalid call sites immediately (by making the code fail to compile), so that they can be fixed ASAP

Yes, it is reasonable, so let's keep current behaviour